snuba/web/rpc/trace_item_values.py

```diff
@@ -0,0 +1,111 @@
+import uuid
+from datetime import datetime
+
+from google.protobuf.json_format import MessageToDict
+from sentry_protos.snuba.v1alpha.endpoint_tags_list_pb2 import (
+    AttributeValuesRequest,
+    AttributeValuesResponse,
+)
+
+from snuba.attribution.appid import AppID
+from snuba.attribution.attribution_info import AttributionInfo
+from snuba.datasets.entities.entity_key import EntityKey
+from snuba.datasets.entities.factory import get_entity
+from snuba.datasets.pluggable_dataset import PluggableDataset
+from snuba.query import OrderBy, OrderByDirection, SelectedExpression
+from snuba.query.data_source.simple import Entity
+from snuba.query.dsl import Functions as f
+from snuba.query.dsl import column, literal, literals_array
+from snuba.query.logical import Query
+from snuba.query.query_settings import HTTPQuerySettings
+from snuba.request import Request as SnubaRequest
+from snuba.utils.metrics.timer import Timer
+from snuba.web.query import run_query
+from snuba.web.rpc.common import base_conditions_and, treeify_or_and_conditions
+from snuba.web.rpc.exceptions import BadSnubaRPCRequestException
+
+
+def _build_query(request: AttributeValuesRequest) -> Query:
+    if request.limit > 1000:
+        raise BadSnubaRPCRequestException("Limit can be at most 1000")
+
+    entity = Entity(
+        key=EntityKey("spans_str_attrs"),
+        schema=get_entity(EntityKey("spans_str_attrs")).get_data_model(),
+        sample=None,
+    )
+
+    # this table stores timestamp as toStartOfDay(x) in UTC, so if you request 4PM - 8PM on a specific day, nada
+    start_timestamp = datetime.utcfromtimestamp(request.meta.start_timestamp.seconds)
+    end_timestamp = datetime.utcfromtimestamp(request.meta.end_timestamp.seconds)
+    if start_timestamp.day == end_timestamp.day:
+        start_timestamp = start_timestamp.replace(
+            day=start_timestamp.day - 1, hour=0, minute=0, second=0, microsecond=0
+        )
+        end_timestamp = end_timestamp.replace(day=end_timestamp.day + 1)
+        request.meta.start_timestamp.seconds = int(start_timestamp.timestamp())
+        request.meta.end_timestamp.seconds = int(end_timestamp.timestamp())
+
+    res = Query(
+        from_clause=entity,
+        selected_columns=[
+            SelectedExpression(
+                name="attr_value", expression=column("attr_value", alias="attr_value")
+            ),
+        ],
+        condition=base_conditions_and(
+            request.meta,
+            f.equals(column("attr_key"), literal(request.name)),
+            # multiSearchAny has special treatment with ngram bloom filters
+            # https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree#functions-support
+            f.multiSearchAny(
+                column("attr_value"),
+                literals_array(None, [literal(request.value_substring_match)]),
+            ),
```

Comment:
Not sure what happens when `request.value_substring_match` is empty but when that happens we should just return any result. Perhaps this condition can be optional

Reply:
if it's empty this condition is a no-op since every string contains the empty string